nexus/Freqlog/Freqlog.py

```diff
@@ -163,47 +170,52 @@ def _log_and_reset_word(min_length: int = 2) -> None:
                     continue
 
                 # Handle whitespace/disallowed keys
-                if ((isinstance(key, kbd.Key) and key in {kbd.Key.space, kbd.Key.tab, kbd.Key.enter}) or
-                        (isinstance(key, kbd.KeyCode) and (not key.char or key.char not in self.allowed_chars))):
+                if (key != '' and key != '\0') and (key in " \t\n\r" or not key or key not in self.allowed_chars):
                     # If key is whitespace/disallowed and timing is more than chord_char_threshold, log and reset word
                     if (word and avg_char_time_after_last_bs and
                             avg_char_time_after_last_bs > timedelta(milliseconds=self.chord_char_threshold)):
                         logging.debug(f"Whitespace/disallowed, log+reset: {word}")
                         _log_and_reset_word()
                     else:  # Add key to chord
-                        match key:
-                            case kbd.Key.space:
-                                word += " "
-                            case kbd.Key.tab:
-                                word += "\t"
-                            case kbd.Key.enter:
-                                word += "\n"
-                            case _:
-                                if isinstance(key, kbd.KeyCode) and key.char:
-                                    word += key.char
+                        if (key != '' and key != '\0'):
+                            word += key
                         last_key_was_disallowed = True
                     self.q.task_done()
                     continue
 
                 # On non-chord key, log and reset word if it exists
                 #   Non-chord key = key in modifier keys or non-key
                 # FIXME: support modifier keys in chords
-                if key in self.modifier_keys or not (isinstance(key, kbd.Key) or isinstance(key, kbd.KeyCode)):
+                if not (key != '' and key != '\0'):
                     logging.debug(f"Non-chord key: {key}")
                     if word:
                         _log_and_reset_word()
                     self.q.task_done()
                     continue
 
+                attributes_mods = [
+                        a for a in dir(vinput.KeyboardModifiers)
+                        if not a.startswith('_') and type(getattr(vinput.KeyboardModifiers, a)).__name__ == "CField"
+                    ]
+                banned_modifier_active = False
+                for attr in attributes_mods:
+                    should_check: bool = getattr(self.modifier_keys, attr)
+                    if not should_check:
```

Comment:
Only used once and quite cryptic, inline it and use a more descriptive comment to describe what it's doing instead.